<launch>
<!-- ============ Arguments ============ -->
  <arg name="data_path" default="/home/lukas/Documents/Datasets/flat_dataset/run1"/>
  <arg name="play_rate" default="1"/>
  <arg name="wait_time" default="0"/>
  <arg name="namespace" default="data"/>
  <arg name="use_detectron" default="false"/>
  <arg name="global_frame_name" default="world"/>
  <arg name="sensor_frame_name" default="depth_camera"/>

    
   
<!-- ============ Play Data ============ -->
   <node pkg="panoptic_mapping_utils" type="flat_data_player.py" name="flat_data_player" output="screen">
    <param name="data_path" value="$(arg data_path)" /> 
    <param name="play_rate" value="$(arg play_rate)" /> 
    <param name="use_detectron" value="$(arg use_detectron)" /> 
    <param name="global_frame_name" value="$(arg global_frame_name)" /> 
    <param name="sensor_frame_name" value="$(arg sensor_frame_name)" /> 
    <param name="wait_time" value="$(arg wait_time)" /> 
            
    <remap from="~color_image" to="$(arg namespace)/color_image" />  
    <remap from="~depth_image" to="$(arg namespace)/depth_image" />  
    <remap from="~id_image" to="$(arg namespace)/segmentation_image" />  
    <remap from="~labels" to="$(arg namespace)/segmentation_labels" />  
    <remap from="~pose" to="$(arg namespace)/pose" />  
  </node>

</launch>
